lib/apytypes/_array_functions.py

```diff
@@ -2167,6 +2169,281 @@ def fullrange(
+        Lastly, negative fixed-point values are represented using the minus sign to match the expectations of VUnit.
+
+    .. versionadded:: 0.4
+
+    Parameters
+    ----------
+    a : :class:`APyFixedArray` or :class:`APyFloatArray`
+        The array to store the bit values of.
+    fname : :class:`str`
+        Path to the CSV file to store bit values.
+    delimiter : :class:`str`, optional
+        Character used in the CSV file to separate bit values. Default is `,`.
+    layout : :class:`str`, optional
+        Layout to use when storing the array. Can be "default" or "vunit".
+
+    See Also
+    --------
+    import_csv
+
+    Examples
+    --------
+    >>> import apytypes as apy
+    >>> a = apy.fx(range(4), int_bits=4, frac_bits=0).reshape((2, 2))
+    >>> apy.export_csv(a, "input.csv")
+
+    A 3-D VUnit array can be exported as
+
+    >>> b = apy.fx(range(8), int_bits=4, frac_bits=0).reshape((2, 2, 2))
+    >>> apy.export_csv(b, "input_vunit.csv", layout="vunit")
+
+    """
+
+    if type(a) in (APyCFixedArray, APyCFloatArray):
+        raise ValueError("Complex data types are not supported yet")
+
+    if layout not in ("default", "vunit"):
+        raise ValueError(f"Unknown layout: {layout}")
+
+    if a.ndim > 3 or (a.ndim > 2 and layout != "vunit"):
+        raise ValueError(f"Unsupported array dimension {a.ndim} for layout {layout}")
+
+    to_bits_fn = (
+        "_to_signed_bits"
+        if type(a) is APyFixedArray and layout == "vunit"
+        else "to_bits"
+    )
+
+    if layout == "vunit":
+        if a.ndim == 2:
+            a = a.T
+        elif a.ndim == 3:
+            a = a.transpose((1, 0, 2))
+
+    with Path(fname).open(mode="w", newline="") as f:
+        writer = csv.writer(f, delimiter=delimiter)
+        if a.ndim == 1:
+            writer.writerow(getattr(a, to_bits_fn)())
+        elif a.ndim == 2:
+            for row in a:
+                writer.writerow(getattr(row, to_bits_fn)())
+        elif a.ndim == 3:
+            for depth in a:
+                writer.writerow(getattr(depth.flatten(), to_bits_fn)())
+
+
+@overload
+def import_csv(
+    delimiter: str = ",",
+    int_bits: int | None = None,
+    frac_bits: int | None = None,
+    bits: int | None = None,
+    exp_bits: int | None = None,
+    man_bits: int | None = None,
+    bias: int | None = None,
+    force_complex: bool = False,
+    layout: str = "default",
+    vunit_3d_shape: tuple[int, ...] | None = None,
+) -> APyFixedArray: ...
+
+
+@overload
+def import_csv(
+    delimiter: str = ",",
+    int_bits: int | None = None,
+    frac_bits: int | None = None,
+    bits: int | None = None,
+    exp_bits: int | None = None,
+    man_bits: int | None = None,
+    bias: int | None = None,
+    force_complex: bool = False,
+    layout: str = "default",
+    vunit_3d_shape: tuple[int, ...] | None = None,
+) -> APyFloatArray: ...
+
+
+def import_csv(
+    fname: str,
+    delimiter: str = ",",
```

Comment:
```suggestion
    *,
    delimiter: str = ",",
```
To make all later argument keyword only. (No point in passing those as positional.)